### ISSUE TITLE
Google Play Music Plug-in

### ISSUE THREAD

**Issue by stayxnegative** [svbnet/Athame#151]
It's no longer working, please fix. [log20180525.log](https://github.com/svbnet/Athame/files/2039957/log20180525.log)

**Comment by lzmichael** [comment]
maybe it works if you have a valid android id, and not computer mac-adress that it works in future.

**Comment by fabriccioguzman** [comment]
not working the plugin

**Comment by DubbyHarry** [comment]
Agree to fabriccioguzman

**Comment by redreadnut** [comment]
guzman, don't panic.

**Comment by jonaaa20** [comment]
It's Guzman after all, he did the same thing with athame plugin... He's desperate without a reason...

**Comment by fabriccioguzman** [comment]
@jonaaa20 I'm not desperate, I was just remembering and confirming that the plugin stopped working

**Comment by FatKiwi** [comment]
You repeat 10 times in 1 sentence 

>  m4a download support from google music, since this is the format that google uses in its streaming service
because this format is the one that uses google and not mp3

and random bullshit

> M4A AAC has more quality than mp3, you could use ffmpeg to encode it and taglibsharp.dll

M4A is better, so transcode it again to mp3 with FFMPEG... WTF Fabriccio?

**Comment by fabriccioguzman** [comment]
@lzmichael @stayxnegative @DubbyHarry @redreadnut @jonaaa20 @FatKiwi  am sorry for having published on m4a several times, but because you did not like my comment, I wanted you to add a format and quality option in the google play music plugin like tidal, m4a is the mp3 successor and even the creators of mp3 already considered obsolete and do not give more licenses
https://www.telegraph.co.uk/technology/2017/05/15/creators-mp3-declare-dead/
![image](https://user-images.githubusercontent.com/27593140/40625675-84b04786-6279-11e8-96a0-6b5eb729af52.png)

**Comment by TripleFrequency** [comment]
@fabriccioguzman 
I agree that mp3 is pretty much dead, but it is illogical to simply convert it to m4a.

You won't gain anything by using m4a, in-fact you will lose quality even further.
If Google Play Music actually encoded using AAC, it would make sense, but you are asking for options to convert one lossy format to another lossy format. Google stores all their GPM music as mp3 @ 320kbps at their highest quality. They even [re-encode any music you upload](https://support.google.com/googleplaymusic/answer/1100462?hl=en) to be mp3. At the end of the day, it's a problem that has been solved by a million other applications, and @svbnet shouldn't need to worry about it here. Perhaps look into [foobar2000](https://www.foobar2000.org/) if you want to do this.

> devs ignore the end user, devs treat end users badly as trash

I don't think you realize the amount of effort and passion that goes into this project, and ones like it. "Simple" features can be very complicated to actually add, especially given that Joe has said before that he doesn't have much time for it. The addition of music transcoding doesn't add any value that makes it worth while, time-wise.

I would advise that you keep this in mind: you aren't a customer to please, ultimately you are getting a service you personally value *for free*. This is at the expense of @svbnet himself, through time/effort, and likely even personal monetary expenses.

If you feel so strongly that your voice isn't heard, perhaps consider [donating to him](https://ko-fi.com/A0343RZ9), and work on presenting your ideas in a more amicable manner. When making suggestions to a project like this, you should always prepare yourself for the possibility that your idea is fully rejected, or placed far back on the backlog.

**Comment by fabriccioguzman** [comment]
@TripleFrequency @jonaaa20 @FatKiwi @svbnet 
ok I'm sorry, I'm sorry, I thought Google Play Music used AAC M4A as audio quality in its streaming service and it turns out that it uses mp3 as a deezer, I apologize to everything and I'll close a topic about this that I have open in the repository of plugin
I apologize for everything, I was the one who made the mistake
As I said, I will soon donate

**Comment by TripleFrequency** [comment]
@fabriccioguzman Don't feel pressured to donate for any reason, only do it for your own reasons. I agree with the idea of moving the issue to the respective repository, though.

I do wish, to a degree, that you hadn't removed your previous message. I feel it removes the important context in which I responded to you, but I understand why you would prefer to remove it.

**Comment by redreadnut** [comment]
Actually Guzman is right. For web streaming google use (or used, I don't know how this reralized today) AAC. I was able to download songs in this format. It was in mp3 but in fact it was AAC, which was hard to play or convert somewhere.

**Comment by fabriccioguzman** [comment]
@TripleFrequency @jonaaa20  @redreadnut @FatKiwi @svbnet 
deleted the comment because I do not want to insist that I implement m4a aac in the add-on and I do not want to talk to someone who will not implement my application, but I am hopeful that I will join AAC but soon I will donate to help the developers and the repository to last almost all life

**Comment by TripleFrequency** [comment]
I think you may be correct about them using AAC for web streaming, but it isn't possible that you downloaded AAC audio inside an MP3 file. MP3, as described directly in its specification, can only store MP3 codec data. M4A is a container, which *can* store MP3 or AAC, along with ~7 other codecs.

I will note this, for @svbnet in particular. I would hold off on any hard work on GPM, and wait to see what happens with the new [YouTube Music](https://music.youtube.com/coming-soon/), which is meant to be like a successor to GPM. It should be rolling out publicly within the coming weeks and months.

**Comment by fabriccioguzman** [comment]
@TripleFrequency I'm confused again, Google Play Music uses mp3 or AAC with the M4a container?

**Comment by TripleFrequency** [comment]
They use the MP3 codec inside an MP3 file for uploaded music and for mobile clients. They use the AAC codec seemingly inside an AAC file for web streaming, as indicated by the player information:
![image](https://user-images.githubusercontent.com/6166013/40634124-8a3ffc08-62c1-11e8-85fc-917d712cf679.png)

**Comment by fabriccioguzman** [comment]
@TripleFrequency mmmm, that's why I said that they implemented AAC with m4a as container with audio quality selector in the plugin but nobody supports me, could it? and also improve the cover quality if possible, will you support me?

**Comment by TripleFrequency** [comment]
I understand the desire for it, but I think currently the plugin uses a mobile client, which only has MP3. Not saying it is impossible, but it is definitely no easy thing.

**Comment by fabriccioguzman** [comment]
currently there is an app for windows called songspout that previously downloaded mp3 and now downloads aac m4a but stopped working but uses ffmpeg to encode and tag lib sharp to write metadata or tags and covers
http://songspout.com/
![image](https://user-images.githubusercontent.com/27593140/40635097-0a581802-62be-11e8-975a-e5d99975ea4f.png)

**Comment by jonaaa20** [comment]
@fabriccioguzman okay, got your point, but just like @TripleFrequency said, you should respect the dev space, he is doing everything for free, you should understand it, but no problems i know that you can't live without music... :grinning:

**Comment by fabriccioguzman** [comment]
@jonaaa20 do not understand that I will not insist more and also I'm not claiming something that is free, it was just a request but since no one supports me, let's leave it like that, now it's up to joe to implement aac with m4a with a web session as I explain @TripleFrecuency

**Comment by svbnet** [comment]
The plugin is based off of a library last updated in 2016, so I understand at some point they would deprecate and remove some aspect of their API. Unfortunately it's much more difficult to reverse engineer as the Android app uses certificate pinning so I can't just inspect traffic between the app and server.

There is the gmusicapi Python project which I can probably port to .NET - this will take a while, and as @TripleFrequency said, Google are planning to roll out a new music service which will probably supersede GPM.

**Comment by fabriccioguzman** [comment]
@svbnet I agree with you and it takes all the time you need but when they replace GPM and if the new music service that will launch google and if they get to launch it with some other format like flac or aac (m4a) or some other, try to include these formats in your plugin.

**Comment by FatKiwi** [comment]
@svbnet this can help you in future: 
https://github.com/ac-pm/SSLUnpinning_Xposed 
or even better https://github.com/ac-pm/Inspeckage

**Comment by Sheepdisease** [comment]
I am encountering this problem.

**Comment by nrafaelos** [comment]
Windows 7 64 bit. Upgrade to Athame-0.4.8.0. Cannot download anything! "an error occurred while downloading this track!'' (Google play music). What am I doing wrong?

**Comment by theluckyman** [comment]
I actually use Athame-0.4.8 so it still issue no music download this time?

**Comment by joselcaguilar** [comment]
Maybe we could solve this, adding the DeviceId as param in the auth?

Just like [mopidy](https://github.com/mopidy/mopidy-gmusic) (written in Python) or other projects updated recently with the last changes from Google.

**Comment by svbnet** [comment]
I'm pretty sure the library I use adds the DeviceId param. I might get it to use the web API (the python library gmusicapi has an excellent implementation).

**Comment by coman3** [comment]
heyo @svbnet, the developer of the Google.Music.Api here... I haven't spent much time looking for a fix mostly because i hadn't noticed the library was in use, and due to work / spare time restraints. Ill have a look into updating the library entirely and adding support for .net standard 2.0 (potentially youtube music as well). For anyone looking for progress updates on this issue, ill do my best to keep the API's repo updated at https://github.com/coman3/Google.Music/tree/dotnet-standard_rework

**Comment by athata** [comment]
@coman3 thanks for your interest, but are you fine with helping fellas downloading music outside Google checkout page?

**Comment by coman3** [comment]
@athata  no im not particularly fine with downloading the songs and having them used maliciously. However, I am fine with people using downloaded music from google play all access; if they are only using the files for their personal gain while ensuring that they stream the files when applicable. (this way the authors can still make their profits)

**Comment by anorexicseal** [comment]
@coman3  i feel the same way. That is why i am still iffy on releasing my version of athame. Since it grabs MQA unencrypted.
